### DISCUSSION TITLE
Cannot reference Region when creating a new Site

### DISCUSSION THREAD

**Discussion by davama** [netbox-community/netbox#7977] [Q&A] [answered]
### NetBox version

v3.0.11

### Python version

3.8.8

### Steps to Reproduce

```
# setup brand new vm with rockylinux8
# gave it internet access via our proxy (http_proxy / https_proxy)
# configure yum.conf with proxy
sed -i 's/=enforcing/=disabled/' /etc/selinux/config 
dnf clean all
dnf update -y
reboot
# at this point i follow the netbox.readthedocs
sudo yum install -y postgresql-server
sudo postgresql-setup --initdb
# edit ident to md5
vi /var/lib/pgsql/data/pg_hba.conf
sudo systemctl start postgresql
sudo systemctl enable postgresql
sudo -u postgres psql
CREATE DATABASE netbox;
CREATE USER netbox WITH PASSWORD 'netbox';
GRANT ALL PRIVILEGES ON DATABASE netbox TO netbox;
\q
sudo yum install -y redis
sudo systemctl start redis
sudo systemctl enable redis
# install python38
dnf install -y python38 
sudo yum install -y gcc libxml2-devel libxslt-devel libffi-devel libpq-devel openssl-devel redhat-rpm-config
pip3 install --upgrade pip
# clone the repo
sudo mkdir -p /opt/netbox/
cd /opt/netbox/
sudo yum install -y git
git clone -b master --depth 1 https://github.com/netbox-community/netbox.git .
git checkout master
git pull origin master
sudo groupadd --system netbox
sudo adduser --system -g netbox netbox
sudo chown --recursive netbox /opt/netbox/netbox/media/
cd /opt/netbox/netbox/netbox/
sudo cp configuration.example.py configuration.py
python3 ../generate_secret_key.py
# ALLOWED_HOSTS = ['*']  
# updated DATABASE with netbox/netbox credentials
# added SECRET_KEY
vi configuration.py 
/opt/netbox/upgrade.sh
source /opt/netbox/venv/bin/activate
cd /opt/netbox/netbox
python3 manage.py createsuperuser # create credentials admin/admin
sudo ln -s /opt/netbox/contrib/netbox-housekeeping.sh /etc/cron.daily/netbox-housekeeping
systemctl stop firewalld
python3 manage.py runserver 0.0.0.0:8000 --insecure
# access my URL http://10.x.x.x:8000/
# create a region called TEST
# attempt to create site under region TEST but 'No Results' comes up when typing TEST under Region field
```

### Expected Behavior

I should be able to create the new Site under TEST Region.

### Observed Behavior

I destroyed my vm and reran the steps above but with v2.11.12 instead, I have no issue creating my first region and creating my first site under said region.

Any input is much appreciated!

EDIT: updated this post, hopefully it's better now

Thank you,
Dave

**Comment by DanSheps** [comment]
Converted to a discussion from #7836

**Comment by davama** [comment]
@DanSheps thank you for doing that.

**Comment by davama** [comment]
Im open to any suggestions/thoughts.
Thank you for the support!

**Comment by candlerb** [comment, marked as answer]
What do you see when you click on Organization > Regions (`/dcim/regions/`) ?  Is your TEST region shown?

When you try to add a new site (`/dcim/sites/add/`), what *exactly* happens when you click on the Region field?  Can you show a screenshot of the dropdown after you click on the field but before you start typing anything?

I am wondering whether you have a problem with AJAX in general.  Try opening your browser's developer tools, then refresh the page (still on `/dcim/sites/add/`), and click on the Region field.  Is any error show in the javascript console?

**Reply by davama** [reply]
@candlerb thank you for your reply

> What do you see when you click on Organization > Regions (`/dcim/regions/`) ? Is your TEST region shown?

Yes, under `/dcim/regions/` my `DAVAMA-REGION` is shown along with other regions. This `DAVAMA-REGION` i just created right now for this Discussion post.
Please see [link](https://imgur.com/a/BIVR8BK)
 


> When you try to add a new site (`/dcim/sites/add/`), what _exactly_ happens when you click on the Region field? Can you show a screenshot of the dropdown after you click on the field but before you start typing anything?

What exactly happens? 
The small drop down text box shows along with the circle arrow on the side and a `Filter` textbox to start typing. But there is only `---` under the `Filter` textbox even when clicking on the circle arrow. When I start typing `D` for the region, the `---` replaces with `No Results`.

See same [link](https://imgur.com/a/BIVR8BK) with scrreenshot. As instructed I did NOT type anything, only clicked.


> 
> I am wondering whether you have a problem with AJAX in general. Try opening your browser's developer tools, then refresh the page (still on `/dcim/sites/add/`), and click on the Region field. Is any error show in the javascript console?

I think we stumbling on something here! Hope I did this right
See [link](https://imgur.com/a/NU3x84H)

I tried the above under Microsoft Edge and there was NO ISSUE!!! I was able to see the Region i created. :100: 
So under Google Chrome I have this issue.... :thinking: 
My Google Chrome Browser is `Version 81.0.4044.138 (Official Build) (64-bit)` on my remote machine... 

On my local machine using Google Chrome `Version 96.0.4664.45 (Official Build) (64-bit)` I see NO ISSUE! :100: 
Also on my local machine with Firefox `94.0.2 (64-bit)` i see NO ISSUE! :100: 

So this is definitely a browser client issue.

Im going to guess that updating the browser will solve it.
But for my educational purpose, is the AJAX problem the issue I am seeing or is it just old browser? or both? @candlerb 

Thank you again for the support!!

**Reply by candlerb** [reply]
> is the AJAX problem the issue I am seeing or is it just old browser? or both?

I'd say they are two sides of the same coin.  Your browser is too old to run Netbox's front-end Javascript code successfully, and it's that code which makes the AJAX calls.

**Reply by davama** [reply]
@candlerb got it
Thank you for the help and input!!!
marking solved

**Reply by DanSheps** [reply]
Chrome 81 is from April 2020 I think,  While I don't know why it couldn't run it, I imagine there are some built in security features that prevent it.

**Reply by candlerb** [reply]
The [screenshot](https://imgur.com/a/NU3x84H) shows it's trying to call Javascript function "replaceAll".  I expect a bit of polyfill could make it work.

Was also reported [here](https://groups.google.com/g/netbox-discuss/c/wj0Jy2Q_1S8/m/TOb9PgY5AwAJ) and in #7142.
